components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -3081,13 +3151,42 @@ private FederatedAuthenticator createFederatedAuthenticator(String authenticator
                     federatedAuthenticator.setTags(Arrays.asList(tags));
                 }
             }
-            List<org.wso2.carbon.identity.api.server.idp.v1.model.Property> properties =
-                    Arrays.stream(config.getProperties()).map(propertyToExternal).collect(Collectors.toList());
-            federatedAuthenticator.setProperties(properties);
+
+            if (DefinedByType.SYSTEM == config.getDefinedByType()) {
+                federatedAuthenticator.setDefinedBy(FederatedAuthenticator.DefinedByEnum.SYSTEM);
+                List<org.wso2.carbon.identity.api.server.idp.v1.model.Property> properties =
+                        Arrays.stream(config.getProperties()).map(propertyToExternal).collect(Collectors.toList());
+                federatedAuthenticator.setProperties(properties);
+            } else {
+                federatedAuthenticator.setDefinedBy(FederatedAuthenticator.DefinedByEnum.USER);
+                resolveEndpointConfiguration(federatedAuthenticator, config);
+            }
+
         }
         return federatedAuthenticator;
     }
 
+    private void resolveEndpointConfiguration(FederatedAuthenticator authenticator,
+                        FederatedAuthenticatorConfig config) throws IdentityProviderManagementServerException {
+
+        try {
+            UserDefinedFederatedAuthenticatorConfig userDefinedConfig =
+                    (UserDefinedFederatedAuthenticatorConfig) config;
+            UserDefinedAuthenticatorEndpointConfig endpointConfig = userDefinedConfig.getEndpointConfig();
+
+            AuthenticationType authenticationType = new AuthenticationType();
+            authenticationType.setType(AuthenticationType.TypeEnum.fromValue(endpointConfig.getEndpointConfig()
+                    .getAuthentication().getType().toString()));
+
+            Endpoint endpoint = new Endpoint();
+            endpoint.setUri(endpointConfig.getEndpointConfig().getUri());
+            authenticator.setEndpoint(endpoint);
+        } catch (ClassCastException e) {
+            throw new IdentityProviderManagementServerException("Error occurred while resolving endpoint " +
+                    "configuration of the authenticator.", e);
```

Comment:
Can we specify the authenticator id  and name ?

Reply:
Addressed.